expr/functions.go

```diff
@@ -0,0 +1,63 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/brimsec/zq/zng"
+)
+
+type Function func([]NativeValue) (NativeValue, error)
+
+var ErrWrongArgc = errors.New("wrong number of arguments")
+var ErrBadArgument = errors.New("bad argument")
+
+var allFns = []struct {
+	name string
+	fn   Function
+}{
+	{"Math.sqrt", mathSqrt},
+}
+
+var allFnsMap map[string]Function
+var fnsInited = false
+
+func lookupFunction(name string) *Function {
+	if !fnsInited {
```

Comment:
Why not just declar allFnsMap as a map literal instead of allFns and skip this initialization?

Reply:
derp, no good reason.  updated.